src/js/rtc_session.js

```diff
@@ -871,7 +871,7 @@ export default class RtcSession {
         var now = new Date();
         self._sessionReport.sessionStartTime = now;
         self._connectTimeStamp = now.getTime();
-        if (pc && pc.signalingState != 'closed') {
+        if (pc && pc.iceConnectionState != 'closed' && pc.iceConnectionState != 'failed' && pc.iceConnectionState != 'disconnected') {
```

Comment:
We decided that we will be adding unit tests for any changes going forward. So please make sure this change is unit tested.